packages/cli/src/Server.ts

```diff
@@ -484,34 +485,33 @@ class App {
 
 
 		// Creates a new workflow
-		this.app.post(`/${this.restEndpoint}/workflows`, ResponseHelper.send(async (req: express.Request, res: express.Response): Promise<IWorkflowShortResponse> => {
-			const newWorkflowData = req.body;
+		this.app.post(`/${this.restEndpoint}/workflows`, ResponseHelper.send(async (req: ICreateWorkflowRequest, res: express.Response): Promise<IShortWorkflow> => {
+			const incomingData = req.body;
 
-			newWorkflowData.name = newWorkflowData.name.trim();
-			newWorkflowData.createdAt = this.getCurrentDate();
-			newWorkflowData.updatedAt = this.getCurrentDate();
+			const newWorkflow = new WorkflowEntity();
+			newWorkflow.createdAt = this.getCurrentDate(); // TODO: Set at DB level
+			newWorkflow.updatedAt = this.getCurrentDate(); // TODO: Set at DB level
 
-			await this.externalHooks.run('workflow.create', [newWorkflowData]);
+			Object.assign(newWorkflow, incomingData);
+			newWorkflow.name = incomingData.name.trim();
```

Comment:
does name need to be validated before `trim` is applied?

Reply:
This is the workflow name. Validation logic in this PR is for tag names.